tests/Fixtures/Integration/misc/PHP8_3.test

```diff
@@ -0,0 +1,116 @@
+--TEST--
+PHP 8.3 test.
+
+@TODO Fix whitespace inside dynamic const fetch, e.g. `Foo::{  $bar  }`
+@TODO Fix whitespace between const type and const name, e.g. `public const string       TEST = 'Test1';`
+--RULESET--
+{
+    "@PhpCsFixer": true,
+    "@PHP83Migration": true
+}
+--REQUIREMENTS--
+{"php": 80300}
+--EXPECT--
+<?php
+
+// https://wiki.php.net/rfc/arbitrary_static_variable_initializers
+function foo(int $j)
+{
+    static $i = bar($j * $j);
+}
+
+// https://wiki.php.net/rfc/dynamic_class_constant_fetch
+class Foo
+{
+    public const BAR = 'bar';
+}
+$bar = 'BAR';
+function bar(): string
+{
+    return 'BAR';
+}
+$baz = ['bar' => ['bar' => 'BAR']];
+
+echo Foo::{$bar};
+echo Foo::{bar()};
+echo Foo::{$baz['bar']['bar']};
+
+// https://wiki.php.net/rfc/typed_class_constants
+enum E
+{
+    public const string TEST = 'Test1';
+}
+
+trait T
+{
+    public const string TEST = E::TEST;
+}
+
+interface I
+{
+    public const string TEST = E::TEST;
+}
+
+class Foo implements I
+{
+    use T;
+
+    public const string TEST = E::TEST;
```

Comment:
> but specific fixers must be improved (or introduced?) in order to fix what currently fails

where is it failing?

Reply:
It would fail if I put superfluous whitespace, as shown in the attached diffs. I removed these spaces just to make it pass and prove it's not failing on new syntaxes, but there's `@todo` in order to improve fixers to handle it properly.

Reply:
IMHO:
- identify syntax changes for 8.3
- (PR 1) misc/PHP8_3 test with syntax that is not causing any issues
- (PR 2, 3, 4, ...) update misc/PHP8_3 with all syntaxes that are causing issues, and we work on fixes
- (PR Z) bring `@PHP83Migration` that will get merged only after 8.3 syntax is not causing any issues

Reply:
Yes, I believe this PR does what you said, more or less. There is integration test that shows Fixer does not fail on new syntaxes, consequent PRs should "break" the input code and provide fixes so expected code is still correct. In terms of `@PHP83Migration` - I've added empty one as a base for further work, but I believe the only fixer we can introduce is a risky one that would add types to constants. Maybe we could add something that would change `constant('bar::'. $const)` to simply `bar::'. $const`? But maybe we can have empty migrations set for now 🤷‍♂️.

Reply:
one thing is to convert syntax, other is to not make it worse.
before we introduce ruleset, we shall ensure we do not make things worse.

can you show the PR that would present the problems mentioned in TODOs?
if we are not fixing sth that we could fix - it;s OK to continue with ruselet. but if we make things worse, let's figure out how to fix them (and thus, 3 PRs)

Reply:
This is shown on the screenshots attached to the PR's description - it does not make it worse, it just does not remove superfluous whitespaces. So for the sake of this particular PR I removed those whitespaces from the input part so it succeeds _with new syntaxes used_. But currently it can't remove spaces between const type and const name and within curly braces of dynamic const fetch, which needs to be improved.

Reply:
to simplify on my side, would you mind to add full expected input and expected output for the part that is not yet working perfectly, as separated commit / subPR ?
The code and CI speaks more to me than the description ;)

Reply:
I believe it's a waste of time since you will see the same diff in the PHPUnit's output as you can already see in attached screenshot, that are taken in my local CLI. But I'll try to do it later, now I'm AFK until evening - family time 🙂.

Reply:
@keradus check #7444's CI to see what fails.

Reply:
It helped me to understood the case and conclude on this (7439) PR, thanks!